gcloud/storage/connection.py

```diff
@@ -169,7 +169,11 @@ def __iter__(self):
         return iter(_BucketIterator(connection=self))
 
     def __contains__(self, bucket_name):
-        return self.lookup(bucket_name) is not None
+        try:
+            self.get_bucket(bucket_name)
+            return True
```

Comment:
I don't have a sense for what is or isn't clean in this instance. Happy to change.